demo/api.html

```diff
@@ -1,11 +1,212 @@
 <div class="container bs-docs-container">
   <div class="row">
     <div class="col-md-9" role="main">
+
       <div class="bs-docs-section">
-        <h1 id="basics" class="page-header">
-          Events
+        <h1 id="routers" class="page-header">
+          Routers
+          <a href="routers" class="anchorjs-link"><span class="anchorjs-icon"></span></a>
+        </h1>
+      </div>
+
+      <div class="bs-docs-section">
+        <h1 id="rules" class="page-header">
+          Rules
+          <a href="rules" class="anchorjs-link"><span class="anchorjs-icon"></span></a>
         </h1>
         <p>
+          Rules are objects that contain a regular expression to match the URL, it might be directly related to the
+          router object or being child of another rule.
+        </p>
+        <p>
+          When a URL change or is manually triggered, the rules are executed in order to check all changes that will
+          occur on the change of the current URL. Those changes create a queue of events to be executed that respect
+          the event priority order.
```

Comment:
I mean, if a regex not match, the child that are in the match state, will dispatch `leave` without executing regexp.